.github/workflows/coverity.yml

```diff
@@ -0,0 +1,164 @@
+          conda list
+
+      - name: Download coverity
+        if: steps.cache-coverity.outputs.cache-hit != 'true'
+        run: |
+          cd
+          curl -X POST https://scan.coverity.com/download/linux64 -d "token=$COV_TOKEN" --data-urlencode "project=MLIR Extensions" -o coverity.tgz
+          tar zxf coverity.tgz
+          mv -T cov-analysis-linux64-* coverity
+
+      - name: Add coverity to PATH
+        run: |
+          echo "$HOME/coverity/bin" >> $GITHUB_PATH
+
+      - name: Show coverity version
+        run: |
+          coverity --version
+
+      - name: Add coverity inputs to env
+        run: |
+          version="$(git rev-parse --short HEAD)"
+          email="alexei.fedotov@intel.com"
+          project_id=30755
+
+          echo "project_id=$project_id" | tee -a $GITHUB_ENV
+          echo "email=$email" | tee -a $GITHUB_ENV
+          echo "version=$version" | tee -a $GITHUB_ENV
+
+      - name: Setup Cache Vars
+        run: |
+          echo 'LLVM_SHA<<EOF' >> $GITHUB_ENV
+          cat $LLVM_SHA_FILE >> $GITHUB_ENV
+          echo 'EOF' >> $GITHUB_ENV
+
+      - name: Cache LLLVM-MLIR
+        id: cache-llvm-mlir
+        uses: actions/cache@v4
+        env:
+          LLVM_CACHE_NUMBER: 1  # Increase to reset cache
+        with:
+          path: |
+            /home/runner/work/llvm-mlir/_mlir_install/**
+          key: ${{ runner.os }}-build-llvm-${{ env.LLVM_CACHE_NUMBER }}-${{ env.LLVM_SHA }}
+
+      - name: Build LLVM-MLIR
+        if: steps.cache-llvm-mlir.outputs.cache-hit != 'true'
+        run: |
+          mkdir -p /home/runner/work/llvm-mlir
+          cd /home/runner/work/llvm-mlir
+          echo "INFO: Need to rebuild LLVM-MLIR. Previous installation for MLIR not found"
+          np=`nproc`
+          echo "INFO: nproc $np"
+          git clone https://github.com/llvm/llvm-project --branch main --single-branch
+          cd llvm-project
+          git checkout ${LLVM_SHA}
+          git apply /home/runner/work/mlir-extensions/mlir-extensions/build_tools/patches/*.patch
+          cmake -G Ninja -B _build -S llvm                                  \
+            -DCMAKE_BUILD_TYPE=MinSizeRel                                  \
+            -DLLVM_ENABLE_PROJECTS=mlir                                    \
+            -DLLVM_ENABLE_ASSERTIONS=ON                                    \
+            -DLLVM_USE_LINKER=gold                                         \
+            -DLLVM_INSTALL_UTILS=ON                                        \
+            -DLLVM_TARGETS_TO_BUILD=X86                                    \
+            -DLLVM_ENABLE_BINDINGS=OFF                                     \
+            -DLLVM_ENABLE_ZSTD=OFF                                         \
+            -DCMAKE_INSTALL_PREFIX=/home/runner/work/llvm-mlir/_mlir_install
+          cmake --build _build --target install
+
+      - name: Build IMEX and run lit tests
```

Comment:
this stage also does coverity build as far as I can see - line 132

Reply:
Thanks, renamed